### PR TITLE
test: Implement known issue throttling

### PR DESCRIPTION
**Pull request by dperpeet**
This reduces bloat on the known issues page and keeps the number
of comments down.

Sample issue: #4048

### CODE REVIEW

File: test/common/testlib.py [hunk 1]
```diff
@@ -731,21 +732,88 @@ def post_github(self, number, err):
...
+            link = "{0}: revision {1}".format(timestamp, revision)
             statuses = github.get("commits/{0}/statuses".format(revision))
             if statuses:
                 for status in statuses:
                     if status["context"] == context:
-                        link = status["target_url"]
+                        link = "[{0}]({1})".format(timestamp, status["target_url"])
+                        break
+
+        comments = github.get("issues/{0}/comments".format(number))
```
Comment (stefwalter) [inline test/common/testlib.py]: Doesn't github page comments? Does this need a loop and ```?page=x&per_page=100```?

Comment (dperpeet) [inline test/common/testlib.py]: > Doesn't github page comments? Does this need a loop and ?page=x&per_page=100?

either that or something else is happening, too: https://github.com/cockpit-project/cockpit/issues/4001#issuecomment-199257031 has duplicate posts

Comment (dperpeet) [inline test/common/testlib.py]: > Doesn't github page comments? Does this need a loop and ?page=x&per_page=100?

fixed

**Comment by dperpeet** [comment]
the script correctly updated [the newest comment](https://github.com/cockpit-project/cockpit/issues/4001#issuecomment-199257031)

File: test/common/testlib.py [hunk 2]
```diff
@@ -731,21 +732,88 @@ def post_github(self, number, err):
         if not github.available:
             return False
 
+        timestamp = datetime.datetime.now().isoformat()
+
         # Lookup the link being logged to
         context = testinfra.DEFAULT_IMAGE
-        link = ""
```
Comment (stefwalter) [inline test/common/testlib.py]: Now that this is a massive function, it should probably move to testinfra.py. A function that updates a known issue in github.

Comment (dperpeet) [inline test/common/testlib.py]: > Now that this is a massive function, it should probably move to testinfra.py. A function that updates a known issue in github.

Done. Specific details, i.e. revision, link to logs are generated in testlib.py, while testinfra.py provides the generic functionality.

**Comment by dperpeet** [comment]
updated https://github.com/cockpit-project/cockpit/issues/4048

**Comment by dperpeet** [comment]
Note that technically this behavior is racy. While one test runner is editing a comment, another might do the same in parallel. The last caller will win and overwrite the first one's changes. Since we're talking about very narrow timeframes (I'd expect <1s) and the error reports are usually longer apart than that, I don't see a problem. Also, a single report being lost on a high frequency issue shouldn't make a big difference.

**Pull request closed without merging**